### PR TITLE
Pending block queue lock fixes

### PR DESCRIPTION
**Pull request by shayzluf**
Bug fix

**What does this PR do? Why is it needed?**
While going through pending block queue locks and unlocks i found some instances where locks are not being released on error. this pr fixes those issues

### CODE REVIEW

**Review by terencechain — DISMISSED** [review]

File: beacon-chain/sync/pending_blocks_queue.go [hunk 1]
```diff
@@ -35,10 +35,10 @@ func (s *Service) processPendingBlocksQueue() {
 	locker := new(sync.Mutex)
 	runutil.RunEvery(s.ctx, processPendingBlocksPeriod, func() {
 		locker.Lock()
+		defer locker.Unlock()
```
Comment (nisdas) [inline beacon-chain/sync/pending_blocks_queue.go:38]: Change this back

**Review by nisdas — CHANGES_REQUESTED** [review]

File: beacon-chain/sync/pending_blocks_queue.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/sync/pending_blocks_queue.go:38]: I missed this. We shouldn't use defer there

Comment (shayzluf) [reply beacon-chain/sync/pending_blocks_queue.go]: reverted

**Review by nisdas — APPROVED** [review]
Nice catches, lgtm

**Pull request merged**